### PR TITLE
Refactored join logic

### PR DESCRIPTION
**Pull request by farhatahmad**
## Description

1- Removed the `join` method and replaced it with `status`
2- Updated mutation to match the new style

### CODE REVIEW

**Review by scouillard — APPROVED** [review]

**Comment by scouillard** [comment]
I remember "understanding" why they were both needed - did you figure out what happened? 

Also, why not name it "join" instead of "status"?
I feel it is more in line with what it is doing?

File: app/javascript/components/rooms/RoomJoin.jsx [hunk 1]
```diff
@@ -6,24 +6,17 @@ import {
...
 import Spinner from '../shared/stylings/Spinner';
 import useRoomStatus from '../../hooks/queries/rooms/useRoomStatus';
 import Avatar from '../users/Avatar';
 
-function waitOrJoin(refetchJoin, refetchStatus) {
-  refetchStatus();
-  refetchJoin();
-}
-
 export default function RoomJoin() {
```
[the remark this answers is not in the document]
Comment (KH-Amir-TN) [reply app/javascript/components/rooms/RoomJoin.jsx:13]: The extraction of the this logic into a separate form component that uses the form framework should answer all of the issues listed above.

File: app/javascript/components/rooms/RoomJoin.jsx [hunk 2]
```diff
@@ -88,7 +81,7 @@ export default function RoomJoin() {
                         />
                       </label>
                       {
-                        (isErrorJoin || isErrorStatus)
+                        (isErrorStatus)
```
[the remark this answers is not in the document]
Comment (KH-Amir-TN) [reply app/javascript/components/rooms/RoomJoin.jsx]: The use of `react-hook-form` can simplify this [CHECK HERE](https://react-hook-form.com/api/useform/seterror)

[the remark this answers is not in the document]
Comment (KH-Amir-TN) [reply app/javascript/components/rooms/RoomJoin.jsx]: The use of `react-hook-form` can even simplify the `wrong access code` logic [CHECK HERE](https://react-hook-form.com/api/useform/seterror)

[the remark this answers is not in the document]
Comment (farhatahmad) [reply app/javascript/components/rooms/RoomJoin.jsx]: I'll tackle this in another PR

File: app/javascript/components/rooms/RoomJoin.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (farhatahmad) [reply app/javascript/components/rooms/RoomJoin.jsx:13]: I'll tackle this in another PR

File: app/controllers/api/v1/meetings_controller.rb [hunk 3]
```diff
@@ -24,32 +24,15 @@ def start
...
 
         if authorized_as_viewer? || authorized_as_moderator?
-          if authorized_as_moderator?
-            bbb_role = 'Moderator'
-          elsif authorized_as_viewer?
-            bbb_role = 'Viewer'
-          end
+          bbb_role = authorized_as_moderator? ? 'Moderator' : 'Viewer'
+
           data[:joinUrl] = BigBlueButtonApi.new.join_meeting(room: @room, name: params[:name], role: bbb_role) if data[:status]
```
[the remark this answers is not in the document]
Comment (farhatahmad) [reply app/controllers/api/v1/meetings_controller.rb:36]: Hmm if this is the case, we might want to just revert this back to how it was before. I'd rather have a small amount of duplication then deal with the complexity of this

File: app/javascript/hooks/queries/rooms/useRoomStatus.jsx [hunk 4]
```diff
@@ -12,6 +13,8 @@ export default function useRoomStatus(friendlyId, name, accessCode) {
       const response = resp.data.data;
       if (response.status) {
         window.location.replace(response.joinUrl);
+      } else {
+        subscribeToRoom(friendlyId, response.joinUrl);
```
[the remark this answers is not in the document]
Comment (farhatahmad) [reply app/javascript/hooks/queries/rooms/useRoomStatus.jsx]: Why poll when we can use sockets?

**Pull request closed without merging**